presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -9,287 +9,434 @@ Except for the limit on queued queries, when a resource group runs out of a reso
+number of queries. This is particularly useful in multi-tenant environments where you want to ensure that no single user
+or query monopolizes the system resources.
+Resource Consumption
+--------------------
+
+A query belongs to a single resource group, and it consumes resources from that group,
+as well as its parent groups. If a resource group runs out of a certain resource, it does not cause running
+queries to fail. Instead, new queries will be queued until resources become available again.
+Sub-Groups and Query Acceptance
+-------------------------------
+
+Sub-groups allow for hierarchical resource allocation, where each sub-group can have its own resource limits and
+queueing policies.A resource group that accepts queries directly is a leaf group, and it executes queries using
+its allocated resources.
+
+The resource groups and associated selection rules are configured by a manager, which is pluggable.
+Presto resource management can be done in two ways:
+
+* File-Based Resource Management
+* Database-Based Resource Management
+
+File-Based Resource Management
+------------------------------
+
+In a file-based resource manager, configuration information about the resource groups is stored in a JSON file.
+This file contains definitions of all resource groups and the rules to select them.
+The configuration file is loaded and used at the start of the Presto server.
+Any changes to the file require a restart of the Presto server to take effect.
+
+Database-Based Resource Management
+----------------------------------
+
+In a database-based resource manager, configuration information about the resource groups is stored in a relational
+database. The database contains definitions of all resource groups and the rules to select them.
+Unlike the file-based resource manager, changes to the configuration in the database take effect
+immediately and do not require a restart of the Presto server.
+
+Both methods have their pros and cons. File-based resource management is simpler to set up but less flexible,
+while database-based resource management is more complex to set up but offers more flexibility and dynamic changes.
+
+File Resource Group Manager
+---------------------------
+
+The File Resource Group Manager in PrestoDB is a way to manage resources using a JSON configuration file.
+This file contains definitions of all resource groups and rules for selecting the appropriate resource group
+for a given query.
+
+To set up a File-based Resource Group Manager:
+
+1. Create the file ``etc/resource-groups.properties``
+2. Set the ``resource-groups.configuration-manager`` property to ``file``.
+3. Create a JSON file with the definitions of your resource groups and rules.
+4. Specify the location of this file in the Presto server's configuration file
+   under the property ``resource-groups.config-file``.
+5. To complete the setup, follow the steps in File Resource Group Manager.
```

Comment:
Where is this step instructing the reader to go? Please review your text for a logical flow.